src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java

```diff
@@ -165,21 +165,57 @@ protected Object decode(
                         if (BitUtil.check(alarm, 2)) {
                             position.addAlarm(Position.ALARM_FALL_DOWN);
                         }
+                        if (BitUtil.check(alarm, 4)) {
+                            position.addAlarm(BitUtil.check(alarm, 26) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 1);
+                        }
+                        if (BitUtil.check(alarm, 5)) {
+                            position.addAlarm(BitUtil.check(alarm, 27) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 2);
+                        }
+                        if (BitUtil.check(alarm, 6)) {
+                            position.addAlarm(BitUtil.check(alarm, 28) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 3);
+                        }
+                        if (BitUtil.check(alarm, 7)) {
+                            position.addAlarm(BitUtil.check(alarm, 29) ? Position.ALARM_GEOFENCE_ENTER : Position.ALARM_GEOFENCE_EXIT);
+                            position.set("geo", 4);
+                        }
                         if (BitUtil.check(alarm, 8)) {
                             position.addAlarm(Position.ALARM_POWER_OFF);
                         }
                         if (BitUtil.check(alarm, 9)) {
                             position.addAlarm(Position.ALARM_POWER_ON);
                         }
+                        if (BitUtil.check(alarm, 10)) {
+                            position.addAlarm(Position.ALARM_MOVEMENT);
+                            position.set(Position.KEY_MOTION, true);
+                        }
+                        if (BitUtil.check(alarm, 11)) {
+                            position.set(Position.KEY_MOTION, false);
+                        }
                         if (BitUtil.check(alarm, 12)) {
                             position.addAlarm(Position.ALARM_SOS);
                         }
+                        if (BitUtil.check(alarm, 13)) {
+                            position.set("side1", true);
+                        }
+                        if (BitUtil.check(alarm, 14)) {
+                            position.set("side2", true);
```

Comment:
What are these?

Reply:
The documentation refers to "Side1 Key" and "Side2 Key". Not sure what they do. 
I'm fine with removing them.

Reply:
Let's remove if we don't know.